karapace/schema_registry_apis.py

```diff
@@ -302,31 +298,27 @@ def send_delete_subject_message(self, subject, version):
     async def compatibility_check(self, content_type, *, subject, version, request):
         """Check for schema compatibility"""
         body = request.json
-        self.log.info("Got request to check subject: %r, version_id: %r compatibility", subject, version)
-        old = await self.subject_version_get(content_type=content_type, subject=subject, version=version, return_dict=True)
-        self.log.info("Existing schema: %r, new_schema: %r", old["schema"], body["schema"])
+        schema_type = self._validate_schema_type(content_type=content_type, data=body)
```

Comment:
Moved the schema type validation up. So that if an error occurs, the logs about the schemas are not printed